tfjs-backend-webgpu/perf/tune.js

```diff
@@ -155,70 +130,137 @@ function initPage() {
   run();
 }
 
-async function timeMatmul(rowNum) {
-  await tf.setBackend('webgpu');
-  const dimAOuter = parseInt(STATE.input.split(',')[0]);
-  const dimInner = parseInt(STATE.input.split(',')[1]);
-  const dimBOuter = parseInt(STATE.input.split(',')[2]);
-  const numIterations = STATE.iteration;
-  const tensorA = tf.tensor2d(
-    Array.from({ length: dimAOuter * dimInner }, () => Math.floor(Math.random())),
-    [dimAOuter, dimInner]
+async function timeMatmul(backend, rowNum) {
+  await tf.setBackend(backend);
+  let tensors = [];
+  let tensorsWarmUp = [];
+  let tensorsToDispose = [];
+  let inputs = [];
+
+  // Prepare data
+  const warmA = tf.tensor2d(
+    Array.from({ length: 512 * 512 }, () => Math.floor(Math.random())),
+    [512, 512]
   );
-  const tensorB = tf.tensor2d(
-    Array.from({ length: dimInner * dimBOuter }, () => Math.floor(Math.random())),
-    [dimInner, dimBOuter]
+  const warmB = tf.tensor2d(
+    Array.from({ length: 512 * 512 }, () => Math.floor(Math.random())),
+    [512, 512]
   );
-  document.getElementById('message').innerHTML =
-    `Testing on webgpu, ${STATE.name}, ${STATE.input} ...`;
-
-  // skip the first time
-  await tf.matMul(tensorA, tensorB).data();
+  tensorsWarmUp = { tensorA: warmA, tensorB: warmB };
+  tensorsToDispose.push(warmA);
+  tensorsToDispose.push(warmB);
+  if (rowNum !== undefined) {
+    let input = document.getElementById(`input-${rowNum}`).getAttribute('value');
+    let m = input.split('[')[1].split(',')[0];
+    let k = input.split('[')[2].split(']')[0].split(',')[0];
+    let n = input.split('[')[2].split(']')[0].split(',')[1];
+    inputs = [`${m},${k},${n}`];
+  }
+  else {
+    inputs = INPUTS;
+  }
+  for (let i = 0; i < inputs.length; i++) {
+    let dimAOuter = parseInt(inputs[i].split(',')[0]);
+    let dimInner = parseInt(inputs[i].split(',')[1]);
+    let dimBOuter = parseInt(inputs[i].split(',')[2]);
```

Comment:
Why we wrap the `dimAOuter | dimInner | dimBOuter` into a string here, and should encode and decode the string? Why not wrap this value into an array or a structure?

Reply:
Each row has a `rerun` button to help rerun a single case for double check. And this button catches shape infos(e.g. [16, 512, 16]) from HTML element, which value must be a string. So here I used the raw data what I get from the page.